core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -69,7 +70,7 @@ abstract class QueryTest extends PlanTest {
       case d: Number               => d.doubleValue()
       case d: String               => BigDecimal(d).doubleValue()
       case d: Boolean              => if (d) 1d else 0d
-      case _                       => 0.0
+      case _                       => Double.NaN
```

Comment:
why change to `Double.NaN`?

Reply:
which means values that cannot be converted to double should not be compared directly

Reply:
In such case, a comment is nice to have.